ToolBox/LeanDataWriter.cs

```diff
@@ -472,13 +481,21 @@ private void WriteFile(string filePath, string data, DateTime date)
             // Create the directory if it doesnt exist
             Directory.CreateDirectory(Path.GetDirectoryName(filePath));
 
-            // Write out this data string to a zip file
-            Compression.Zip(data, tempFilePath, LeanData.GenerateZipEntryName(_symbol, date, _resolution, _tickType));
-
-            // Move temp file to the final destination with the appropriate name
-            File.Move(tempFilePath, filePath);
+            if (_appendToZips)
+            {
+                var entryName = LeanData.GenerateZipEntryName(_symbol, date, _resolution, _tickType);
+                Compression.ZipCreateAppendData(filePath, entryName, data);
```

Comment:
`Compression.ZipCreateAppendData` won't update the entry if it already exists and log an error and return `false`. Believe this can happen if calling `Write(IEnumerable<BaseData> source)`, they don't seem to be removing the entry if it exists as `DownloadAndSave()` is, this seems in part to the code duplication we have here 😅  - Maybe we can add an override argument to `Compression.ZipCreateAppendData`. Thoughts @Jay-Jay-D